src/screencast/pipewire_screencast.c

```diff
@@ -13,6 +13,35 @@
 #include "xdpw.h"
 #include "logger.h"
 
+static bool pwr_y_invert_buffer(struct pw_buffer *buffer) {
+	uint32_t N;
+	struct spa_data *d = buffer->buffer->datas;
+	void *tmp_buffer = malloc(d->chunk->stride);
+	if (!tmp_buffer) {
+		return false;
+	}
+	void *data = mmap(NULL, d[0].maxsize, PROT_READ | PROT_WRITE, MAP_SHARED,
+			d[0].fd, d[0].mapoffset);
+	if (!data) {
+		free(tmp_buffer);
+		return false;
+	}
+	uint32_t height = d->chunk->size / d->chunk->stride;
```

Comment:
There's no guarantee the height will be divisible by the stride, and that the result is the height…